balloc/balloc.c

```diff
@@ -3045,19 +3080,29 @@ M0_INTERNAL int m0_balloc_create(uint64_t              cid,
 			rc = -ENOMEM;
 		} else {
 			cb->cb_container_id = cid;
+			M0_BE_ALLOC_CREDIT_ARR(cb->cb_db_group_extents,
+					       bcfg->bfr_indexcount, seg, &cred);
+			M0_BE_ALLOC_ARR_SYNC(cb->cb_db_group_extents,
+					     bcfg->bfr_indexcount, seg, &tx);
+			if(cb->cb_db_group_extents == NULL)
+				return M0_RC(-ENOMEM);
 
 			balloc_format_init(cb);
-			m0_be_btree_init(&cb->cb_db_group_extents, seg,
-					 &ge_btree_ops);
+			for (i = 0; i < bcfg->bfr_indexcount; i++) {
+				m0_be_btree_init(&cb->cb_db_group_extents[i], seg,
+						 &ge_btree_ops);
+			}
 			m0_be_btree_init(&cb->cb_db_group_desc, seg,
 					 &gd_btree_ops);
-			rc = balloc_trees_create(cb, &tx, fid);
+			rc = balloc_trees_create(cb, &tx, &bcfg->bfr_fid,
+						 bcfg->bfr_indexcount);
 			if (rc == 0) {
 				M0_BE_TX_CAPTURE_PTR(seg, &tx, cb);
 				*out = cb;
 			} else {
-				m0_be_btree_fini(&cb->cb_db_group_extents);
-				m0_be_btree_fini(&cb->cb_db_group_desc);
```

Comment:
Why is "m0_be_btree_fini(&cb->cb_db_group_desc);"  removed?

Do we need to free `cb->cb_db_group_extents'?

Reply:
It was removed during  branch migration, added back again.